### PR TITLE
ergo

### PR DESCRIPTION
**Pull request by isovector**

### CODE REVIEW

File: src/Weft/Generics/AllTypes.hs [hunk 1]
```diff
@@ -1,73 +1,73 @@
...
-instance (HasAllTypes record) => GFindTypes (K1 _1 [record 'Data]) where
-  gFindTypes = findTypes @record
-
-instance (HasAllTypes record) => GFindTypes (K1 _1 (NonEmpty (record 'Data))) where
-  gFindTypes = findTypes @record
-
-instance {-# OVERLAPPABLE #-} GFindTypes (K1 _1 _2) where
-  gFindTypes = id
+module Weft.Generics.AllTypes where
+  --( HasAllTypes
```
Comment (JonathanLorimer) [inline src/Weft/Generics/AllTypes.hs]: Do we still need this?

File: src/Weft/Generics/Hydrate.hs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module Weft.Generics.Hydrate
-  ( HasHydrate
-  , hydrate
-  , hydrateF
+  ( HasMagicHydrate
+  , magicHydrate
+  -- , hydrateF
```
Comment (JonathanLorimer) [inline src/Weft/Generics/Hydrate.hs]: Can we add hydrateF back?

File: src/Weft/Generics/Hydrate.hs [hunk 3]
```diff
@@ -12,59 +12,116 @@ import           Weft.Internal.Types hiding (query)
...
+             (J record 'Query)
+             (J record 'Response)
   )
 
 
 ------------------------------------------------------------------------------
 -- |
-hydrate :: HasHydrate record => record 'Data -> record 'Query -> record 'Response
-hydrate d query = to $ gHydrate (from d) (from query)
+magicHydrate :: HasMagicHydrate record => record -> JHKD record 'Query -> JHKD record 'Response
```
Comment (JonathanLorimer) [inline src/Weft/Generics/Hydrate.hs:25]: can we just call it hydrate?

File: src/Weft/Generics/Hydrate.hs [hunk 4]
```diff
@@ -12,59 +12,116 @@ import           Weft.Internal.Types hiding (query)
...
+instance (Generic record, GHydrate (Rep record)
+                  (M1 _1 _2 fq)
+                  (M1 _1 _2 fr)) =>
+      GHydrateTerm [record]
+                   (M.Map Text (Args args, M1 _1 _2 fq _4))
+                   (M.Map Text [M1 _1 _2 fr _4]) where
+  gHydrateTerm d q = fmap ((<$> fmap from d) . flip gHydrate . snd) q
+
+
+-- instance (Generic record, GHydrate (Rep record)
```
Comment (JonathanLorimer) [inline src/Weft/Generics/Hydrate.hs]: can this be removed?

File: src/Weft/Generics/JSONResponse.hs [hunk 5]
```diff
@@ -1,10 +1,11 @@
 module Weft.Generics.JSONResponse
-  ( HasJSONResponse
-  , jsonResponse
+  ( HasMagicJSONResponse
+  , magicJsonResponse
```
Comment (JonathanLorimer) [inline src/Weft/Generics/JSONResponse.hs:3]: Isn't the user going use this function? shouldn't it be called `jsonResponse`

File: src/Weft/Generics/QueryParser.hs [hunk 6]
```diff
@@ -5,13 +5,11 @@
...
 module Weft.Generics.QueryParser
-  ( HasQueryParser
-  , HasMagicQueryParser
+  ( HasMagicQueryParser
   , Vars
   , Parser
-  , queryParser
   , magicQueryParser
-  , anonymousQueryParser
+  -- , anonymousQueryParser
```
Comment (JonathanLorimer) [inline src/Weft/Generics/QueryParser.hs:12]: Can we get rid of this?

File: src/Weft/Generics/QueryParser.hs [hunk 7]
```diff
@@ -34,30 +32,21 @@ import           Weft.Internal.ParserUtils
...
-magicQueryParser = lift skipCrap *> gQueryParser <* lift skipCrap
 
-
-anonymousQueryParser :: HasQueryParser q => ReaderT Vars Parser (Gql q m s 'Query)
-anonymousQueryParser = do
-  r <- parens '{' '}' queryParser
-  pure $ Gql { query    = M.singleton "query" (ANil, r)
-             , mutation = M.empty
-             }
+-- anonymousQueryParser :: HasQueryParser q => ReaderT Vars Parser (Gql q m s 'Query)
```
Comment (JonathanLorimer) [inline src/Weft/Generics/QueryParser.hs:44]: Same with this

File: test/AllTypesSpec.hs [hunk 8]
```diff
@@ -1,27 +1,30 @@
 module AllTypesSpec where
```
Comment (JonathanLorimer) [inline test/AllTypesSpec.hs]: should delete this too

**Review by JonathanLorimer — CHANGES_REQUESTED** [review]
left a couple comments

File: src/Weft/Generics/Hydrate.hs (hunk 3 above)
Comment (isovector) [reply src/Weft/Generics/Hydrate.hs:25]: yes but later!

File: src/Weft/Generics/JSONResponse.hs (hunk 5 above)
Comment (isovector) [reply src/Weft/Generics/JSONResponse.hs:3]: yes but later

File: src/Weft/Generics/QueryParser.hs (hunk 6 above)
Comment (isovector) [reply src/Weft/Generics/QueryParser.hs:12]: this still needs to exist! let me rewrite it

**Review by JonathanLorimer — APPROVED** [review]

**Pull request merged**